Framework/Algorithms/src/CalculateMultipleScattering.cpp

```diff
@@ -539,28 +541,19 @@ std::tuple<bool, double, double> CalculateMultipleScattering::scatter(
 void CalculateMultipleScattering::q_dir(Geometry::Track &track, const MatrixWorkspace_sptr SOfQ, const double kinc,
                                         const double scatteringXSection, Kernel::PseudoRandomNumberGenerator &rng,
                                         double &QSS, double &weight) {
-  auto qvalues = SOfQ->histogram(0).x().rawData();
-  // For elastic just select a q value in range 0 to 2k
-  // The following will eventually be used for inelastic where it's less trivial
-  double QQ, cosT;
-  bool foundQ = false;
-  for (auto m = 0; m < 1000; m++) {
-    QQ = qvalues.back() * rng.nextValue();
-    // T = 2theta
-    cosT = 1 - QQ * QQ / (2 * kinc * kinc);
-    if (abs(cosT) <= 1) {
-      foundQ = true;
-      break;
-    }
-  }
-  if (!foundQ) {
-    throw std::runtime_error("Unable to select a new q for kinc=" + std::to_string(kinc));
-  } else {
-    double SQ = interpolateLogQuadratic(SOfQ, QQ);
-    QSS += QQ * SQ;
-    weight = weight * scatteringXSection * SQ * QQ;
-    updateTrackDirection(track, cosT, rng.nextValue() * 2 * M_PI);
-  }
+  double QQ, cosT, qrange, kf;
```

Comment:
Can I nitpick on good practice of defining variables at the same time as assigning them so we can make them `const`, i.e.
```
const double kf = kinc;
const double qrange = 2 * kinc
```
It avoids undefined variables and also prevents you from accidentally using assigning to the wrong variable later
on.